python/pyarrow/array.pxi

```diff
@@ -17,8 +17,8 @@
 
 
 cdef _sequence_to_array(object sequence, object mask, object size,
-                        DataType type,
-                        CMemoryPool* pool, c_bool from_pandas):
+                        DataType type, CMemoryPool* pool, c_bool from_pandas,
+                        c_bool safe):
```

Comment:
TODO: Implementation is missing from here

Reply:
ConvertPySequence [assumes](https://github.com/apache/arrow/blob/master/cpp/src/arrow/python/python_to_arrow.cc#L926) strict conversion if datatype is explicitly passed.